### PR TITLE
IA-2195 cross build service test

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-2195

Gather some feedback about this approach...

The goal is cross publish serviceTest to 2.13 so that we're unblocked from upgrading leonardo to 2.13. ServiceTest module is the last dependency we need to upgrade.

Since serviceTest module relies on `rawlsModel`, which is only available for scala 2.12. We have 2 options if we still want to cross build serviceTest
1. duplicate rawls code in wb-libs
2. cross build rawls-model...

I'd prefer to defer the decision on option 2 to teams maintaining rawls. Hence I'm taking option 1 in this PR.

cc @davidangb @kshakir 

Test https://github.com/broadinstitute/rawls/pull/1290

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/AttributeUpdateOperations.scala [hunk 1]
```diff
@@ -0,0 +1,277 @@
+package org.broadinstitute.dsde.workbench.service
+
+// This file is entirely copied from https://github.com/broadinstitute/rawls/blob/develop/model/src/main/scala/org/broadinstitute/dsde/rawls/model/AttributeUpdateOperations.scala#L9
```
Comment (kshakir) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/AttributeUpdateOperations.scala]: Can you change this to a GitHub Permalink? `develop` line number nine is likely to change or even disappear in the future.

**Comment by davidangb** [comment]
@Qi77Qi is there a third option, in which `AttributeUpdateOperations.scala` moves (not copies) into workbench-libs, and then Rawls references those classes from workbench-libs?

**Comment by Qi77Qi** [comment]
@davidangb ah yea, I guess that's another option...I can try moving this data type to `workbenchModel`, and I'll leave it to you guys to adopt the class in rawls?

**Comment by davidangb** [comment]
@Qi77Qi sure, we could take on updating Rawls to adopt the class from here.

**Comment by Qi77Qi** [comment]
@kshakir @davidangb think I addressed all comments so far...let me know if you guys have any other concerns

**Review by kshakir — APPROVED** [review]

**Review by davidangb — APPROVED** [review]
sorry for the delay - lgtm. I haven't tried this in Rawls yet.

**Comment by Qi77Qi** [comment]
@davidangb thanks for reviews! do you know of any automation tests using https://github.com/broadinstitute/workbench-libs/blob/4de4ed0cd2e0647e3b779552fece5be76e2f37e8/serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Rawls.scala#L277? I'm happy to try running the test with this version just to be sure

**Comment by Qi77Qi** [comment]
@davidangb think I'm going to merge by EOD today...happy to run a test still and fix things up if there's any issue

**Comment by davidangb** [comment]
@Qi77Qi it looks like these tests use that method:

https://github.com/broadinstitute/rawls/blob/develop/automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala#L214

https://github.com/broadinstitute/rawls/blob/develop/automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspaceApiSpec.scala#L229

**Comment by Qi77Qi** [comment]
@davidangb oh hmm..didn't realize these tests are just in rawls repo...thought somewhere else are using these hence we have these helpers in wb-libs...anyway, thanks for the reference, I'll test out a published version of this code

**Comment by davidangb** [comment]
I haven't checked to see if they are used anywhere else, I just looked in the Rawls repo ... it's possible they are used elsewhere too!

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GKEInterpreter.scala [hunk 2]
```diff
@@ -47,13 +47,13 @@ final class GKEInterpreter[F[_]: Async: StructuredLogger: Timer: ContextShift](
         Async[F].delay(clusterManagerClient.getCluster(clusterId.toString)),
         whenStatusCode(404)
       ),
-      f"com.google.cloud.container.v1.ClusterManagerClient.getCluster(${clusterId.toString})"
+      s"com.google.cloud.container.v1.ClusterManagerClient.getCluster(${clusterId.toString})"
```
Comment (kshakir) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GKEInterpreter.scala:54]: Thinking-out-Loud (ToL): While you're here, `.toString` isn't needed.

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GKEInterpreter.scala [hunk 3]
```diff
@@ -47,13 +47,13 @@ final class GKEInterpreter[F[_]: Async: StructuredLogger: Timer: ContextShift](
...
     )
 
   override def deleteCluster(clusterId: KubernetesClusterId)(implicit ev: ApplicativeAsk[F, TraceId]): F[Operation] =
     tracedGoogleRetryWithBlocker(
       Async[F].delay(clusterManagerClient.deleteCluster(clusterId.toString)),
-      f"com.google.cloud.container.v1.ClusterManagerClient.deleteCluster(${clusterId.toString})"
+      s"com.google.cloud.container.v1.ClusterManagerClient.deleteCluster(${clusterId.toString})"
```
Comment (kshakir) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GKEInterpreter.scala]: Same re: `.toString`

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GKEInterpreter.scala [hunk 4]
```diff
@@ -77,7 +77,7 @@ final class GKEInterpreter[F[_]: Async: StructuredLogger: Timer: ContextShift](
         Async[F].delay(clusterManagerClient.getNodePool(nodepoolId.toString)),
         whenStatusCode(404)
       ),
-      f"com.google.cloud.container.v1.ClusterManagerClient.getNodepool(${nodepoolId.toString})"
+      s"com.google.cloud.container.v1.ClusterManagerClient.getNodepool(${nodepoolId.toString})"
```
Comment (kshakir) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GKEInterpreter.scala:94]: Again

**Pull request merged**